worlds/witness/hints.py

```diff
@@ -338,22 +340,85 @@ def get_priority_hint_locations(world: "WitnessWorld") -> List[str]:
     return priority
 
 
+def try_getting_location_group_other_world(world: "WitnessWorld", hint_loc: Location) -> str:
+    possible_location_groups = {
+        k: v for k, v in world.multiworld.worlds[hint_loc.player].location_name_groups.items()
+        if hint_loc.name in v
+    }
+
+    locations_in_that_world = {
+        location.name for location in world.multiworld.get_locations(hint_loc.player) if location.address is not None
+    }
+
+    valid_location_groups = dict()
+
+    for group, locations in possible_location_groups.items():
+        present_locations = sum(location in locations_in_that_world for location in locations)
+        if present_locations > 1:
+            valid_location_groups[group] = present_locations
+
+    if len(valid_location_groups) > 1 and valid_location_groups["Everywhere"] > 100:
+        del valid_location_groups["Everywhere"]
+
+    parent_region = hint_loc.parent_region
```

Comment:
Just to make sure, are you okay with the hint saying that an item is in something like "MAP_PETALBURG_CITY_SUPER_ROD_ENCOUNTERS" or "REGION_MOSSDEEP_CITY_SPACE_CENTER_1F/MAIN"? Region names aren't necessarily designed for players to read during gameplay.

Reply:
If this becomes a regular occurance or issue for players, then I would change the feature. However, right now I find the idea of this happening to someone every once in a while very funny. Lol

Reply:
For Emerald, the majority of locations are in regions that only have one or two locations. The majority of remaining locations aren't in any location groups. So it's very likely you'd just get "Everywhere" or the region name almost every time. And I was slightly wrong in my example, the dexsanity region would be "REGION_POKEDEX" in all cases.

We've also been pushing the "regions can be conceptual, not just physical" idea in the dev channels lately, which could be a strange mix here if you hit a large region of physically distant locations locked behind a common item.

Just wanted to bring it up though. Player testing will reveal the feel of it better than speculation I'm sure.

Reply:
Had a bit of a thought and I think I'm making "region names as fallback" optional calling that version of it `experimental mode` with a warning in the option tooltip.

Reply:
Sounds fine to me.